### DISCUSSION TITLE
Switch between terminal outputs

### DISCUSSION THREAD

**Discussion by mrphil2105** [mfussenegger/nvim-dap#1169] [Q&A]
Hey. I am using `nvim-dap` together with `nvim-dap-ui` to debug a monorepo project. I am running multiple debug sessions at once, which results in an arbitrary output of a random debug configuration being shown in the DAP Console. I need a way to switch between the different session outputs. I have tried looking through the source code of both `nvim-dap` and `nvim-dap-ui`, but without any luck. It seems `nvim-dap-ui` supplies a single buffer number to `nvim-dap`, which is then used to display output inside the `on_stdout` function and send inputs in the `on_input` function, using the `nvim_chan_send` function. Is there perhaps a way I can switch between which stdout and stdin is being sent to the channel? Thanks in advance.

**Comment by mfussenegger** [comment]
Is this about the REPL or about integrated terminals?
If it's the REPL, can you switch to integrated terminals?
If it's integrated terminal - with pure nvim-dap you'd get one dedicated terminal window per debug session. If that's not supported in nvim-dap-ui you'd have to bring it up there.

**Reply by mrphil2105** [reply]
It is about integrated terminals. I know I can create a buffer per debug session, e.g. by using tabnew. But it seems nvim-dap-ui always provides the same buffer number, regardless of how many debug sessions I have. Is there possibly a way to create x hidden buffers and somehow sync them with the nvim-dap-ui buffer, and switch between which one is synced? Or would I have to add multiple buffer support to nvim-dap-ui directly?

**Reply by mfussenegger** [reply]
>  But it seems nvim-dap-ui always provides the same buffer number, regardless of how many debug sessions I have.

Then you have to ask for help in the nvim-dap-ui project. nvim-dap uses a buffer pool for the terminals:

https://github.com/mfussenegger/nvim-dap/blob/fc880e82059eb21c0fa896be60146e5f17680648/lua/dap/session.lua#L170-L201
